### ISSUE TITLE
[.Net][Template] [SelectAction is doubled in the JSON object]

### ISSUE THREAD

**Issue by arman-g** [microsoft/AdaptiveCards#3977]
# Platform

* .NET Templating SDK

# Version of SDK

NuGet 1.2.4

# Details

Adding **SelectAction** to a **AdaptiveContainer** or **AdaptiveColumnSet** adds **SelectAction** to the AdaptiveCard (parent element) also. It basically duplicates the **SelectAction** (please see the image below).

![Select Action Bug](https://user-images.githubusercontent.com/22778273/80838886-de65b800-8bae-11ea-98f3-a00a74add409.png)

### C# Code

```
Body.Add(new AdaptiveContainer
{
	Spacing = AdaptiveSpacing.Padding,
	Items = new List<AdaptiveElement>
	{
		new AdaptiveColumnSet
		{
			SelectAction = SelectAction = new AdaptiveToggleVisibilityAction
			{
				TargetElements = new List<AdaptiveTargetElement>
				{
					new AdaptiveTargetElement
					{
						ElementId = "malfeasance_explained",
						IsVisible = true
					},
					new AdaptiveTargetElement
					{
						ElementId = "show_info_icon",
						IsVisible = false
					}
				}
			},
			Columns = new List<AdaptiveColumn>
			{
				new AdaptiveColumn
				{
					Width = "stretch",
					Items = new List<AdaptiveElement>
					{
						new AdaptiveTextBlock
						{
							Text = messages[1126],
							Wrap = true
						}
					}
				},
				new AdaptiveColumn
				{
					Id = "show_info_icon",
					VerticalContentAlignment = AdaptiveVerticalContentAlignment.Center,
					Width = "auto",
					Items = new List<AdaptiveElement>
					{
						new AdaptiveImage
						{
							HorizontalAlignment = AdaptiveHorizontalAlignment.Center,
							UrlString = _httpContext.GetAbsoluteUrl("info.png"),
							PixelWidth = 16,
							PixelHeight = 16
						}
					}
				}
			}
		}
	}
});
```

**Comment by shalinijoshi19** [comment]
@jwoo-msft another test card for you to test our new templating package out with. Meanwhile @arman-g we are in the middle of publishing an update for our new reworked templating package. It should be out in the next couple of weeks. Please watch the [releases page](https://github.com/microsoft/AdaptiveCards/releases) for updates. We'll make sure your testcard works with our new offering. For now going ahead and closing this out.

**Comment by shalinijoshi19** [comment]
Watch out for #3946 getting merged into master here.

**Comment by jwoo-msft** [comment]
@arman-g 
cc @shalinijoshi19 
I don't think you are having problem with our template library.  The problem description seems applies to .NET AdaptiveCards SDK. Templated string start with '{' and ends with '}' in old syntax.
For example, "text" : "{Hello}"
Here, "{Hello}" is templated string. 

The current published version of the nuget is 0.1.0-alpha1.
https://www.nuget.org/packages/AdaptiveCards.Templating/0.1.0-alpha1

1.2.4 is AdaptiveCards SDK's nuget version.

**Comment by arman-g** [comment]
> @arman-g
> cc @shalinijoshi19
> I don't think you are having problem with our template library. The problem description seems applies to .NET AdaptiveCards SDK. Templated string start with '{' and ends with '}' in old syntax.
> For example, "text" : "{Hello}"
> Here, "{Hello}" is templated string.
> 
> The current published version of the nuget is 0.1.0-alpha1.
> https://www.nuget.org/packages/AdaptiveCards.Templating/0.1.0-alpha1
> 
> 1.2.4 is AdaptiveCards SDK's nuget version.

@jwoo-msft maybe but that does not explain why an instance of AdaptiveCard class gets SelectAction property Instantiated when a child element SelectAction is set.

I have to do this workaround to get rid of that nasty bug :)
```
 public Attachment ToAttachment()
        {
            SelectAction = null; // bug (https://github.com/microsoft/AdaptiveCards/issues/3977) workaround fix.
            return new Attachment
            {
                ContentType = ContentType,
                Name = GetType().Name,
                // Needed to overcome circular reference error in adaptive cards framework.
                Content = this.ConvertTo(typeof(object))
            };
        }
```

**Comment by arman-g** [comment]
@jwoo-msft actually you are right, this is not a bug but a copy and paste honest mistake. Very hard to find bug. The following line in my code sets the image and adaptive card **SelectAction** properties. 
```
SelectAction = SelectAction = new AdaptiveToggleVisibilityAction
{
	TargetElements = targetElements
}
```
